### PR TITLE
New search param for filters

### PR DESCRIPTION
**Pull request by zMotivat0r**
- added new `search` query param to compound `and`, `or` conditions in any variations
- `@nestjsx/crud-request` now builds a query string with [qs](https://www.npmjs.com/package/qs) package
- improvements in `@nestjsx/crud-request` builder and parser

UPDATE:
So after some thoughts, I decided to follow @Diluka [suggestion](https://github.com/nestjsx/crud/pull/265#pullrequestreview-290642508).
So now it's a very flexible API for queries.

### CODE REVIEW

File: packages/crud-request/src/request-query.parser.ts [hunk 1]
```diff
@@ -195,7 +202,29 @@ export class RequestQueryParser implements ParsedRequestParams {
...
-  private conditionParser(cond: 'filter' | 'or', data: string): QueryFilter {
+  private parseSearchQueryParam(data?: any): QuerySearchParsed {
+    let d = !isNil(data) ? data : this._query[this.getParamNames('search')[0]];
+
+    if (isString(d)) {
+      d = this.conditionParser('search', d);
+    }
+
+    if (isArrayFull(d)) {
+      d = d.map((one: string) => this.parseSearchQueryParam(one));
```
Comment (Darkein) [inline packages/crud-request/src/request-query.parser.ts]: you could return d.map directly instead of reassign it

Comment (zMotivat0r) [reply packages/crud-request/src/request-query.parser.ts]: right

**Comment by Darkein** [comment]
that's a nice one, thanks !

File: packages/crud-request/src/request-query.parser.ts [hunk 2]
```diff
@@ -195,7 +202,29 @@ export class RequestQueryParser implements ParsedRequestParams {
...
+    }
+
+    if (isObject(d)) {
+      if (isArrayFull(d.and)) {
+        d.and = this.parseSearchQueryParam(d.and);
```
[the remark this answers is not in the document]
Comment (zMotivat0r) [reply packages/crud-request/src/request-query.parser.ts]: `isArrayFull(p)` - don't think that will work

**Comment by Diluka** [comment]
I was created a common crud controller for `sequelize` 🤦‍♂ 

https://gist.github.com/Diluka/1d326621ce60d63572f58fc42843c518

File: packages/crud-request/src/request-query.parser.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Darkein) [reply packages/crud-request/src/request-query.parser.ts]: sorry, I fixed it

**Comment by zMotivat0r** [comment]
@Diluka, please review

**Pull request merged**